mirgecom/inviscid.py

```diff
@@ -128,32 +128,183 @@ def inviscid_facial_flux(discr, state_tpair, local=False):
+        \right)}{\left(s^+ - s^-\right)}
+
+    where $f^{\mp}$, $q^{\mp}$, and $s^{\mp}$ are the interface-normal fluxes, the
+    states, and the wavespeeds for the interior (-) and exterior (+) of the
+    interface respectively.
+
+    Details about how the parameters and fluxes are calculated can be found in
+    Section 10.3 of [Toro_2009]_.
+    """
+    # calculate left/right wavespeeds
+    actx = state_pair.int.array_context
+    ones = 0.*state_pair.int.mass_density + 1.
+
+    # note for me, treat the interior state as left and the exterior state as right
+    # pressure estimate
+    p_int = state_pair.int.pressure
+    p_ext = state_pair.ext.pressure
+    u_int = np.dot(state_pair.int.velocity, normal)
+    u_ext = np.dot(state_pair.ext.velocity, normal)
+    rho_int = state_pair.int.mass_density
+    rho_ext = state_pair.ext.mass_density
+    c_int = state_pair.int.speed_of_sound
+    c_ext = state_pair.ext.speed_of_sound
+
+    p_star = (0.5*(p_int + p_ext) + (1./8.)*(u_int - u_ext)
+              * (rho_int + rho_ext) * (c_int + c_ext))
+
+    gamma_int = gas_model.eos.gamma(state_pair.int.cv, state_pair.int.temperature)
+    gamma_ext = gas_model.eos.gamma(state_pair.ext.cv, state_pair.ext.temperature)
+
+    q_int = 1 + (gamma_int + 1)/(2*gamma_int)*(p_star/p_int - 1)
+    q_ext = 1 + (gamma_ext + 1)/(2*gamma_ext)*(p_star/p_ext - 1)
+
+    pres_check_int = actx.np.greater(p_star, p_int)
+    pres_check_ext = actx.np.greater(p_star, p_ext)
+
+    q_int = actx.np.where(pres_check_int, q_int, ones)
+    q_ext = actx.np.where(pres_check_ext, q_ext, ones)
+
+    q_int = actx.np.sqrt(q_int)
+    q_ext = actx.np.sqrt(q_ext)
+
+    # left (internal), and right (external) wave speed estimates
+    # can alternatively use the roe estimated states to find the wave speeds
+    s_minus = u_int - c_int*q_int
+    s_plus = u_ext + c_ext*q_ext
+
+    f_minus_normal = inviscid_flux(state_pair.int)@normal
+    f_plus_normal = inviscid_flux(state_pair.ext)@normal
+
+    q_minus = state_pair.int.cv
+    q_plus = state_pair.ext.cv
+
+    from mirgecom.flux import num_flux_hll
+    return num_flux_hll(f_minus_normal, f_plus_normal, q_minus, q_plus, s_minus,
+                        s_plus)
 
-    if local is False:
-        return discr.project(dd, dd_all_faces, flux_weak)
 
-    return flux_weak
+def inviscid_boundary_flux_for_divergence_operator(
+        discr, gas_model, boundaries, interior_boundary_states,
+        domain_boundary_states, quadrature_tag=None,
+        numerical_flux_func=inviscid_flux_rusanov, time=0.0):
+    """Compute the inviscid boundary fluxes for the divergence operator.
+
+    This routine encapsulates the computation of the inviscid contributions
+    to the boundary fluxes for use by the divergence operator. Its existence
+    is intended to allow multiple operators (e.g. Euler and Navier-Stokes) to
+    perform the computation without duplicating code.
```

Comment:
Please have another look @majosm. I tried to fix up the names I hope more to your liking, and maybe describing what they do more accurately. 

I do not disagree with your notion to rename... but just to explain my thinking here; this code in the fluid operators (and the supporting fluid infrastructure) is describing the treatment of a single DG element.  Boundary in that context is just the element boundary regardless of whether there is another element on the other side (local or remote) or the domain boundary.  That's what I mean by interior-boundary vs. domain-boundary.  

The interior boundaries (including those faces on the partition boundaries) are handled with `inviscid_facial_flux` as we have all the info we need (+) and (-) states and use the Riemann solver there.   Domain boundary handled by the BC routine and we have only the (-) side state available as input for those.  The BC is expected to return the numerical flux that will be used for that face.  How the BC gets the flux is the BC's business.  Users are free to implement time-dependent fluxing functions for those - thus the **kwargs so that the other flux routines don't holler when they get a `time` argument or whatever other parameters the user's flux routine might require.